openmc/mgxs/mdgxs.py

```diff
@@ -388,19 +410,53 @@ def get_xs(self, groups='all', subdomains='all', nuclides='all',
 
         # Reshape tally data array with separate axes for domain, energy groups,
         # delayed groups, and nuclides
-        num_subdomains = int(xs.shape[0] / (num_groups * num_delayed_groups))
-        new_shape = (num_subdomains, num_delayed_groups, num_groups)
-        new_shape += xs.shape[1:]
-        xs = np.reshape(xs, new_shape)
+        # Accomodate the polar and azimuthal bins if needed
+        if self.num_polar or self.num_azimuthal:
+            if self.num_polar:
+                num_pol = self.num_polar
+            else:
+                num_pol = 1
+            if self.num_azimuthal:
+                num_azi = self.num_azimuthal
+            else:
+                num_azi = 1
+            num_subdomains = int(xs.shape[0] /
+                                 (num_groups * num_delayed_groups *
+                                  num_pol * num_azi))
+            new_shape = (num_pol, num_azi, num_subdomains, num_delayed_groups,
+                         num_groups) + xs.shape[1:]
+            xs = np.reshape(xs, new_shape)
+
+            # Reverse data if user requested increasing energy groups since
+            # tally data is stored in order of increasing energies
+            if order_groups == 'increasing':
+                xs = xs[:, :, :, :, ::-1, :]
```

Comment:
This can likely be simplified in a more sustainable way as:


```python
if order_groups == 'increasing':
    xs = xs[..., ::-1, :]
```

Perhaps you want to explicitly illustrate each dimension in the expression for transparency, which is also fine.